src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
OK, I had a closer look, and I now have a better understanding of what is happening. We only filter out dynamic assemblies when we [build the list of already loaded assembly files](https://github.com/FakeItEasy/FakeItEasy/blob/a9ba9a42d14db7171c2beac3a04702a4bee8d272/src/FakeItEasy/Core/TypeCatalogue.cs#L87-L96). But we still [scan those assemblies](https://github.com/FakeItEasy/FakeItEasy/blob/a9ba9a42d14db7171c2beac3a04702a4bee8d272/src/FakeItEasy/Core/TypeCatalogue.cs#L101)... Which means my fix doesn't actually fix anything: it just prevents us from loading `DynamicProxyGenAssembly2` again from a file, which is impossible anyway.

I think the only reason we're filtering out dynamic assemblies is because they don't have a `Location`, and accessing that property would cause a `NotSupportedException`. But we don't exclude them otherwise, and we still try to load extension points from them.

Technically, it would be a breaking change to stop doing that. If a user generates extension points with Castle.Core before using FakeItEasy, they will be in `DynamicProxyGenAssembly2 `, and the user will expect them to be loaded. So we can't even filter out `DynamicProxyGenAssembly2`. Admittedly, this is a pretty unlikely scenario... We *could* assume nobody is doing that now, and make it an unsupported scenario. Or we could do nothing, and let the warnings show up sometimes in the integration tests (or we could capture the console output so that they don't show up in the build output).

What do you think?